src/averageAges.js

```diff
@@ -37,7 +54,36 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let peopleNew = [];
+  let count = 0;
+  let mother = '';
+
+  if (withChildren === true) {
+    for (const i of people) {
+      mother = i.mother;
+
+      if (people.find(callback)) {
+        if (peopleNew.find((arr) => (arr.name === mother))) {
+        } else {
+          peopleNew.push({ ...people.find(callback) });
+        }
+      }
+    }
+  } else {
+    peopleNew = people.filter((arr) => (arr.sex === 'f'));
+  }
+
+  function callback(arr) {
+    return (arr.name === mother);
+  };
+
+  function callbackSum(summa, n) {
+    return (summa + n.died - n.born);
+  }
```

Comment:
you can move these functions out of `calculateWomenAverageAge` and reuse them everywhere